tests/SimplifyTest.elm

```diff
@@ -5062,6 +5063,99 @@ a = x
         ]
 
 
+stringSliceTests : Test
+stringSliceTests =
+    describe "String.slice"
+        [ test "should not report String.slice that contains variables or expressions" <|
+            \() ->
+                """module A exposing (..)
+a = String.slice b c
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectNoErrors
+        , test """should replace String.slice b 0 by \"\"""" <|
+            \() ->
+                """module A exposing (..)
+a = String.slice b 0
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Using String.slice with end index 0 will result in an empty string"
+                            , details = [ "You can replace this call by an empty string." ]
+                            , under = "String.slice"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = ""
```

Comment:
This is incorrect: You can only do this when all 3 arguments have been passed to `String.slice`.

```elm
String.slice b 0
--> `always ""`
```
(Not sure whether parentheses are needed here)